src/subcommands/deployments.ts

```diff
@@ -81,6 +119,88 @@ export default async function (args: Args): Promise<void> {
   }
 }
 
+async function listDeployments(args: Args): Promise<void> {
+  if (!args.project) {
+    error(
+      "No project specified. Use --project to specify the project of which to list the deployments",
+    );
+  }
+  const relativeNext = args.next.reduce(
+    (prev, next) => prev + parseInt(next || "1"),
+    0,
+  );
+  const relativePrev = args.prev.reduce(
+    (prev, next) => prev + parseInt(next || "1"),
+    0,
+  );
+  // User-facing page is 1-based. Paging in API is 0-based.
+  const page = parseInt(args.page || "1") + relativeNext - relativePrev;
+  if (page < 1) {
+    error(`The page cannot be lower than 1. You asked for page '${page}'`);
+  }
+  const apiPage = page - 1;
+  const limit = args.limit ? parseInt(args.limit) : undefined;
+  let format: "overview" | "json";
+  switch (args.format) {
+    case "overview":
+    case "json":
+      format = args.format;
+      break;
+    case undefined:
+      format = isTerminal(Deno.stdout) ? "overview" : "json";
+      break;
+    default:
+      error(
+        `Invalid format '${args.format}'. Supported values for the --format option are 'overview' or 'json'`,
+      );
+  }
+  const spinner = wait(
+    `Fetching page ${page} of the list of deployments of project '${args.project}'...`,
+  )
+    .start();
+  const api = args.token
+    ? API.fromToken(args.token)
+    : API.withTokenProvisioner(TokenProvisioner);
+  const [deployments, project, databases] = await Promise.all([
+    api.listDeployments(
+      args.project,
+      apiPage,
+      limit,
+    ),
+    api.getProject(args.project),
+    api.getProjectDatabases(args.project),
+  ]);
+  if (!deployments || !project || !databases) {
+    spinner.fail(
+      `The project '${args.project}' does not exist, or you don't have access to it`,
+    );
+    return Deno.exit(1);
+  }
```

Comment:
It's possible that project does exist but no databases exist associated with that project, right? I guess in such a scenario we'd like to show project info at least?

Reply:
If there is no database, it will return an empty array of databases. The only error that the databases list endpoint returns (controlledly) is `ProjectNotFound`

Reply:
Oh sorry I thought `!foo` where foo is an empty array would be evaluated as `true` but it's not the case actually. I always feel confused about truthy / falsy of values in JavaScript 🤦‍♂️